### ISSUE TITLE
Unspecific warning messages for @todo commands starting with Doxygen 1.8.18

### ISSUE THREAD

**Issue by pkl97** [doxygen/doxygen#7699]
File [testcases.tar.gz](https://github.com/doxygen/doxygen/files/4475331/testcases.tar.gz) demonstrates the warnings emitted by Doxygen 1.8.18:

`testcase1: dd/da0/todo:1: warning: expected <dt> tag but found <p> instead!`
`testcase2: dd/da0/todo:4: warning: unexpected token TK_COMMAND_BS as the argument of _internalref`

Both warnings are related to \@todo statements. The culprit in test case 1 is a misplaced \@todo statement that tries to document the only member of a struct. In test case 2 another \@todo statement is added which leads to a completely different message. In both cases nothing points to the origin of the messages.

If your project is rather large it is a challenge to identify which part Doxygen is complaining about.
Maybe it is possible to add some more information to the messages to aid in debugging.

Regards, Peter.

**Comment by albert-github** [comment]
As you already mentioned the usage of the doxygen comment (testcase1, similar for testcase2):
```
struct MyStruct
{
    PoolNames m_poolNames; /// @todo Change reading of chain elements to make m_poolNames const
};
```
is not correct, it should have been:
```
struct MyStruct
{
    PoolNames m_poolNames; ///< @todo Change reading of chain elements to make m_poolNames const
};
```

With version 1.8.17 it is silently ignored though with 1.8.18 a warning is given:
```
todo:1: warning: expected <dt> tag but found <p> instead!
```
The commit causing the warning or better said not silently ignoring the setting is:
> Commit: 9faa7bdc0d6231e96a83f1f6400a982307fdab2d [9faa7bd]
> Date: Thursday, February 27, 2020 10:43:06 PM
> Restructure the way RefLists are handled

with a small update of
> Commit: 84951f8ee67a7000fe617b73f7e50b9495763ad9 [84951f8]
> Date: Friday, February 28, 2020 12:14:32 AM
> Fix for compilation issue on Linux

I think it is correct that a warning is given although it is very cryptic and it would be nice to have a reference to the original place but the warning is given in a general part of the code (`int DocHtmlDescList::parse()` in `docparser.cpp`) so I'm not sure whether or not this is feasible.

**Comment by albert-github** [comment]
It looks like this item is related to #7738 and the  proposed patch, pull request #7740 

It removes the messages that occur due to a comment that is not used, as other texts are ignored as well when they would have been in such a misplaced comment, I think we can safely say that the proposed patch #7740 also solves this issue.

**Comment by albert-github** [comment]
Code of proposed pull request #7740 has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.19.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
